docs/source/implementations/labeling_matrix_flags.rst

```diff
@@ -0,0 +1,149 @@
+.. _implementations-labeling_matrix_flags:
+
+=====================
+Labeling Matrix Flags
+=====================
+
+The matrix flag labeling method is meant to capture trends in financial prices data, such as bull and bear flags.
+A bull flag occurs when a stock's price rapidly increases, followed by a downwards trending consolidation period, followed by a breakout
+increase in price confirming the original increase. As defined, "A bull flag pattern is a horizontal or downward
+sloping flag of consolidation followed by a sharp rise in the positive direction, the breakout." [Leigh et al. 2002].
+Being able to identify the early stages of the breakout process can lead to a profitable strategy of buying the breakout and
+then selling some number of days later, when the price has theoretically stabilized again.
+
+.. tip::
+   **Literature Reference**
+
+    Labeling price data according to a template to identify patterns in price changes is featured in the following paper:
+    `Cervelló-Royo, R., Guijarro, F. and Michniuk, K., 2015. Stock market trading rule based on pattern recognition and technical
+    analysis: Forecasting the DJIA index with intraday data. <https://www.sciencedirect.com/science/article/abs/pii/S0957417415002110>`_
+
+    This method was originally introduced in `Leigh, W., Modani, N., Purvis, R. and Roberts, T., 2002. Stock market trading rule discovery
+    using technical charting heuristics. <http://chart-patterns.technicalanalysis.org.uk/LMPR02.pdf>`_, which describes this method in more
+    detail.
+
+Cervelló-Royo et al. expand on Leigh et al.'s work by proposing a new bull flag pattern which ameliorates some weaknesses in Leigh's original template,
+such as the possibility of false positives given the path the stock took. Additionally, he applies this bull flag labeling method to intraday
+candlestick data, rather than just closing prices.
+
+The bull flag labeling pattern requires the use of a template to apply to the transformed price data. Below is an example bull flag
+template.
+
+.. figure:: labeling_images/bull_flag_template.png
+   :scale: 60 %
+   :align: center
+   :figclass: align-center
+   :alt: bull flag template
+
+   Bull flag template, from Cervelló-Royo et al. (2015), originally proposed by Leigh et al. (2002). The first 7 columns represent
+   consolidation, while final 3 columns represent the early stages of the breakout.
+
+The matrix flag labeling method, in order to find a single day's fit, consists of the following steps:
+
+ - The data window is taken, which consists of the current day's price and a number of days preceding.
+ - The prices in the data window are ranked and decile cutoffs are found. A mapping is found between each price and its decile.
```

Comment:
I like this algorithm description much better!


`A mapping is found between each price and its decile.` can be changed to `Each price is mapped to a corresponding decile.`